pkg/kubeletconfig/kubeletconfig.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeletconfig
+
+import (
+	"context"
+	"encoding/json"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	mcov1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+)
+
+func MCOKubeletConfToKubeletConf(mcoKc *mcov1.KubeletConfig) (*kubeletconfigv1beta1.KubeletConfiguration, error) {
+	kc := &kubeletconfigv1beta1.KubeletConfiguration{}
+	err := json.Unmarshal(mcoKc.Spec.KubeletConfig.Raw, kc)
+	return kc, err
+}
+
+func GetMCPsFromMCOKubeletConfig(ctx context.Context, cli client.Client, mcoKubeletConfig mcov1.KubeletConfig) ([]*mcov1.MachineConfigPool, error) {
```

Comment:
this belongs more in `pkg/machineconfigpools/` or even `pkgs/machineconfigpools/find`

Reply:
Being the KubeletConfig the information we have, I kind of think of them as the "receiver" of the message to get MCOs from them ... but I do not really have a strong opinion about that.

Reply:
that's a valid point, but the function is about finding MCPs, and the `pkg/machineconfigpools/find` is indeed about finding MCPs using different search criterias. It seems a better fit: from a package named `kubeletconfig` we expect code which manipulates kubeletconfigs.

Reply:
ok. I will change it